### PR TITLE
Refactor russian roulette

### PR DESCRIPTION
**Pull request by tooruu**

### CODE REVIEW

File: arabot/modules/games.py [hunk 1]
```diff
@@ -68,10 +68,9 @@ def _apply_move(self, player: int, column: int) -> int:
     def _check_4_in_a_row(self, last_added: int) -> bool:
         target_value = self._state[last_added]
 
-        space_right = 6 - last_added % 7
-        space_left = last_added % 7
-        space_down = 5 - last_added // 7
-        space_up = last_added // 7
+        space_up, space_left = divmod(last_added, 7)
+        space_right = 6 - space_left
+        space_down = 5 - space_up
```
Comment (tooruu) [inline arabot/modules/games.py:73]: Ignore this.

Comment (AdiReich) [reply arabot/modules/games.py:73]: ignored

File: arabot/modules/games.py [hunk 2]
```diff
@@ -380,50 +379,81 @@ def check_board_winner(self) -> disnake.abc.User | bool:
...
+
+    def reload(self) -> None:
+        if self.mega_killed():
+            self.last_deaths.clear()
+        self.current_pos = 0
+        self.bullet_pos = random.randint(1, self.chambers)
+        self.last_shooter_id = 0
+
+
+def rr_cooldown(msg: disnake.Message) -> commands.Cooldown | None:
```
Comment (AdiReich) [inline arabot/modules/games.py:425]: Make cooldown an argument with default value of 60 so its configurable.

File: arabot/modules/games.py [hunk 3]
```diff
@@ -380,50 +379,81 @@ def check_board_winner(self) -> disnake.abc.User | bool:
...
+    rev = RussianRoulette.GAMES.get(msg.guild.id)
+    if rev and rev.last_shooter_id != msg.author.id and rev.killed():
```
Comment (AdiReich) [inline arabot/modules/games.py]: You always run RussianRoulette.reload()
before 
RussianRoulette.killed()
this means statement will be forever false

File: arabot/modules/games.py [hunk 4]
```diff
@@ -380,50 +379,81 @@ def check_board_winner(self) -> disnake.abc.User | bool:
...
-        )
-
-    def rr_cooldown(self, msg: disnake.Message) -> commands.Cooldown | None:
-        if self.rr_last_user.get(msg.guild.id) != msg.author.id:
-            barrel = self.rr_barrel[msg.guild.id]
-            if barrel[0] == barrel[1]:
-                return commands.Cooldown(1, 60)
-        return None
 
+    @commands.dynamic_cooldown(rr_cooldown, commands.BucketType.guild)
```
Comment (AdiReich) [inline arabot/modules/games.py:438]: @commands.dynamic_cooldown is still buggy
and will break when someone tries to run ;rr command
shortly after dynamic_cooldown has epxired

**Review by AdiReich — CHANGES_REQUESTED** [review]

File: arabot/modules/games.py (hunk 4 above)
Comment (AdiReich) [reply arabot/modules/games.py:438]: I'll close it cus this is disnake issue

**Review by AdiReich — APPROVED** [review]
lgtm

**Pull request merged**